metsrw/mets.py

```diff
@@ -369,48 +382,49 @@ def _get_el_to_normative(parent_elem, normative_parent_elem):
-                'mets:fileSec//mets:file[@ID="' + file_id + '"]',
-                namespaces=utils.NAMESPACES)
-            if file_elem is None:
-                raise exceptions.ParseError(
-                    '%s exists in structMap but not fileSec' % file_id)
-            use = file_elem.getparent().get('USE')
-            path = file_elem.find(
-                'mets:FLocat', namespaces=utils.NAMESPACES).get(
-                    utils.lxmlns('xlink') + 'href')
-            try:
-                path = utils.urldecode(path)
-            except ValueError:
-                raise exceptions.ParseError(
-                    'Value "{}" (of attribute xlink:href) is not a valid'
-                    ' URL.'.format(path))
-            amdids = file_elem.get('ADMID')
-            checksum = file_elem.get('CHECKSUM')
-            checksumtype = file_elem.get('CHECKSUMTYPE')
-            file_id_prefix = utils.FILE_ID_PREFIX
-            # If the file is an AIP, then its prefix is not "file-" but the
-            # name of the AIP. Therefore we need to get the extension-less
-            # basename of the AIP's path and remove its UUID suffix to ge
-            # the prefix to remove from the FILEID attribute value.
-            if entry_type.lower() == 'archival information package':
-                file_id_prefix = os.path.splitext(
-                    os.path.basename(path))[0][:-36]
-            file_uuid = file_id.replace(file_id_prefix, '', 1)
-            group_uuid = file_elem.get('GROUPID', '').replace(
-                utils.GROUP_ID_PREFIX, '', 1)
-            if group_uuid != file_uuid:
-                derived_from = group_uuid  # Use group_uuid as placeholder
-            return FPtr(file_uuid, derived_from, use, path, amdids,
-                        checksum, checksumtype)
+    def _analyze_fptr(fptr_elem, tree, entry_type):
+        file_uuid = derived_from = use = path = amdids = checksum = \
+            checksumtype = None
+        file_id = fptr_elem.get('FILEID')
+        file_elem = tree.find(
+            'mets:fileSec//mets:file[@ID="' + file_id + '"]',
+            namespaces=utils.NAMESPACES)
+        if file_elem is None:
+            raise exceptions.ParseError(
+                '%s exists in structMap but not fileSec' % file_id)
+        use = file_elem.getparent().get('USE')
+        path = file_elem.find(
+            'mets:FLocat', namespaces=utils.NAMESPACES).get(
+                utils.lxmlns('xlink') + 'href')
+        try:
+            path = utils.urldecode(path)
+        except ValueError:
+            raise exceptions.ParseError(
+                'Value "{}" (of attribute xlink:href) is not a valid'
+                ' URL.'.format(path))
+        amdids = file_elem.get('ADMID')
+        checksum = file_elem.get('CHECKSUM')
+        checksumtype = file_elem.get('CHECKSUMTYPE')
+        file_id_prefix = utils.FILE_ID_PREFIX
+        # If the file is an AIP, then its prefix is not "file-" but the
```

Comment:
This whole section kind of scares me 😬. Given way the files are formatted though, I think it has to be like this.

Reply:
It is scary!